### PR TITLE
PLONEOPS-18362 : configure timeout property for keywhiz server

### PR DESCRIPTION
**Pull request by vn0ytch**
### Description

Write here

### Fixes

Changes proposed in this pull request.

### Check List

- [ ] All test passed
- [ ] Added test to ensure to fix/ensure properly.

### CODE REVIEW

File: src/main/java/com/oneops/proxy/config/ApplicationConfig.java [hunk 1]
```diff
@@ -104,19 +104,20 @@ public KeywhizKeyStore ldapKeyStore(OneOpsConfig config, ResourceLoader loader)
 
   /** Returns the keywhiz http client */
   @Bean
-  @Lazy
   public KeywhizClient keywhizClient(
       OneOpsConfig config, @Qualifier("keywhizKeyStore") KeywhizKeyStore keywhizKeyStore)
       throws GeneralSecurityException {
-    return new KeywhizClient(config.getKeywhiz().getBaseUrl(), keywhizKeyStore);
+    OneOpsConfig.Keywhiz keywhiz = config.getKeywhiz();
+    return new KeywhizClient(/*config.getKeywhiz().getBaseUrl(),*/ keywhizKeyStore, keywhiz);
```
Comment (sureshg) [inline src/main/java/com/oneops/proxy/config/ApplicationConfig.java]: Keywhiz config has the keywhiz store. So why do we need to inject both configs?

File: src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java [hunk 2]
```diff
@@ -126,9 +134,9 @@ protected OkHttpClient createHttpsClient() throws GeneralSecurityException {
             .connectionSpecs(singletonList(ConnectionSpec.MODERN_TLS))
             .followSslRedirects(false)
             .retryOnConnectionFailure(true)
-            .connectTimeout(5, SECONDS)
-            .readTimeout(5, SECONDS)
-            .writeTimeout(5, SECONDS)
+            .connectTimeout(keywhiz != null ? keywhiz.getConnectTimeout() : 5, SECONDS)
```
Comment (sureshg) [inline src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java]: keywhiz can't be null as we are using the baseUrl above. Also, the default value shouldn't be in the code. It should be part of the application  yaml config.

File: src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java [hunk 3]
```diff
@@ -126,9 +134,9 @@ protected OkHttpClient createHttpsClient() throws GeneralSecurityException {
...
+            .readTimeout(keywhiz != null ? keywhiz.getReadTimeout() : 5, SECONDS)
```
Comment (sureshg) [inline src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java]: Remove the `!=null` checks

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -64,6 +64,9 @@ oneops:
     svc-user: ${KEYWHIZ_USER}
     svc-password: ${KEYWHIZ_PASS}
     secret-max-size: ${KEYWHIZ_SECRET_MAX_SIZE:350000}
+    connect-timeout: ${KEYWHIZ_CONNECT_TIMEOUT}
```
Comment (sureshg) [inline src/main/resources/application.yaml]: Add the default value here like in max size.

File: src/main/java/com/oneops/proxy/config/ApplicationConfig.java (hunk 1 above)
Comment (vn0ytch) [reply src/main/java/com/oneops/proxy/config/ApplicationConfig.java]: @sureshg : config.getKeywhiz().getKeyStore() returns KeyStore not KeywhizKeyStore.

File: src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java (hunk 2 above)
Comment (vn0ytch) [reply src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java]: Yeah, i will update that now.

File: src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java (hunk 3 above)
Comment (vn0ytch) [reply src/main/java/com/oneops/proxy/keywhiz/http/HttpClient.java]: yes

File: src/main/resources/application.yaml (hunk 4 above)
Comment (vn0ytch) [reply src/main/resources/application.yaml]: ok

Comment (vn0ytch) [reply src/main/resources/application.yaml]: ${KEYWHIZ_CONNECT_TIMEOUT:5}
Like this: ^

File: src/main/java/com/oneops/proxy/config/ApplicationConfig.java (hunk 1 above)
Comment (sureshg) [reply src/main/java/com/oneops/proxy/config/ApplicationConfig.java]: https://github.com/oneops/secrets-proxy/blob/master/src/main/java/com/oneops/proxy/security/KeywhizKeyStore.java#L51
KeywhizKeyStore is created from that config. So you don't need to inject both

File: src/main/resources/application.yaml (hunk 4 above)
Comment (sureshg) [reply src/main/resources/application.yaml]: yes, but the default value should be 10 seconds.

**Pull request merged**